core/sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
+                return transaction.Status.pending
+
+            raise MessageException('operation failed '
```

Comment:
When a transaction exists but has not been committed, transaction_page raise Error(http.FOUND(=302), 'transaction not committed {0}'.format(txnid)), then return error_response that code =  302. The urllib2.open will raise urllib2.HTTPError, and err.code = 302. So client couldn't return transaction.Status.pending if we don't add the line 'if err.code == 302:' there.